packages/components/FocusZone/macos/RCTFocusZone.m

```diff
@@ -45,6 +45,19 @@ static inline CGFloat GetMinDistanceBetweenRectVerticesAndPoint(NSRect rect, NSP
 	);
 }
 
+static NSView *GetFirstKeyViewWithin(NSView *parentView)
+{
+	for (NSView *view in [parentView subviews]) {
+		if ([view canBecomeKeyView]) {
+			return view;
```

Comment:
I haven't looked into Apple's algorithm here but thought I'd ask - do we know that Apple does basically a depth first search through the view hierarchy instead of breadth first to find the first view to use?

Reply:
We actually don't know. The closest answer I could find is here https://cocoadev.github.io/KeyViewLoopGuidelines/

Reply:
Couldn't we just test it? Make a view hierarchy like so [v1: [v2: [v4], v3]] where v2 has canBecomeKeyView set to NO and it's YES for v3 and v4.

I'm not sure how common this would come up, but it would be notable difference worth at least considering even if we end up ignoring it

Reply:
That's fair. I can write a test app, try this out, and update.

Reply:
That's fair. I can write a test app, try this out, and update.